### PR TITLE
Master wowl jpp

### PR DESCRIPTION
**Pull request by jpp-odoo**

### CODE REVIEW

File: addons/web/static/src/search/layout.xml [hunk 1]
```diff
@@ -10,4 +19,19 @@
         </div>
     </t>
 
+    <t t-name="web.DialogLayout" owl="1">
+        <t t-call="web.DialogTemplateFormat">
+            <t t-set="bodyTemplate" t-value="'web.StandardLayout'"/>
+            <t t-set="bodyClass" t-value="{'o_act_window': env.config.actionType === 'ir.actions.act_window'}"/>
```
Comment (kebeclibre) [inline addons/web/static/src/search/layout.xml]: this should not be here

File: addons/web/static/tests/views/form/form_view_tests.js [hunk 2]
```diff
@@ -6700,38 +6700,37 @@ QUnit.module("Views", (hooks) => {
...
+                </form>
+            `,
             resId: 1,
-            viewOptions: {
-                footerToButtons: true,
-                mode: "edit",
-            },
         });
 
+        form.edit();
```
Comment (jpp-odoo) [inline addons/web/static/tests/views/form/form_view_tests.js:6724]: Needed ?

File: addons/web/static/src/views/form/form_view.xml [hunk 3]
```diff
@@ -15,7 +20,7 @@
                             resModel="model.root.resModel"/>
                     </t>
                 </t>
-                <t t-set-slot="control-panel-bottom-right"></t>
+                <t t-set-slot="control-panel-bottom-right"><div/></t>
```
Comment (kebeclibre) [inline addons/web/static/src/views/form/form_view.xml]: doubt about the empty slot....

Comment (kebeclibre) [reply addons/web/static/src/views/form/form_view.xml]: https://github.com/odoo/owl/pull/1169/files

File: addons/web/static/src/views/form/form_view.js [hunk 4]
```diff
@@ -155,6 +155,18 @@ export class FormView extends Component {
         // enable the archive feature in Actions menu only if the active field is in the view
         this.archiveEnabled = "active" in activeFields || "x_active" in activeFields;
 
+        if (this.archInfo.xmlDoc.querySelector("footer")) {
+            const footerXmlDoc = createElement("t");
+            footerXmlDoc.append(...[...this.archInfo.xmlDoc.querySelectorAll("footer")]);
+            this.footerArchInfo = new FormArchParser().parse(
```
Comment (kebeclibre) [inline addons/web/static/src/views/form/form_view.js]: for action actionDialog: maybe only copy archInfo of the whole arch ?

File: addons/web/static/src/views/form/form_view.xml [hunk 5]
```diff
@@ -5,7 +5,12 @@
         <div t-att-class="props.className" t-ref="root">
             <Layout className="model.useSampleModel ? 'o_view_sample_data' : ''">
                 <t t-set-slot="control-panel-bottom-left">
-                    <t t-call="{{ constructor.buttonTemplate }}" />
+                    <t t-if="footerArchInfo and env.inDialog">
```
Comment (kebeclibre) [inline addons/web/static/src/views/form/form_view.xml:8]: simplify if ?

File: addons/web/static/src/views/view_button/view_button.js [hunk 6]
```diff
@@ -46,10 +49,23 @@ export class ViewButton extends Component {
...
     onClick() {
-        this.env.onClickViewButton({
-            clickParams: this.props.clickParams,
-            record: this.props.record,
-        });
+        if (this.props.confirm) {
+            const dialogProps = {
+                body: this.props.confirm,
+                confirm: () =>
+                    this.env.onClickViewButton({
```
Comment (kebeclibre) [inline addons/web/static/src/views/view_button/view_button.js]: all this within onClickViewButton ?
+
make sure the dialog is closed when the view is

File: addons/web/static/tests/views/form/form_view_tests.js [hunk 7]
```diff
@@ -6700,29 +6700,26 @@ QUnit.module("Views", (hooks) => {
         await click(form.el.querySelector(".o_field_x2many_list_row_add a"));
     });
 
-    QUnit.skipWOWL("modifiers are considered on multiple <footer/> tags", async function (assert) {
-        assert.expect(2);
-
-        const form = await makeView({
+    QUnit.test("modifiers are considered on multiple <footer/> tags", async function (assert) {
+        await makeView({
```
Comment (jpp-odoo) [inline addons/web/static/tests/views/form/form_view_tests.js]: Change the test to create a Webclient and call an inline action with target="new"

**Pull request merged**